### PR TITLE
(feature) Move Settings to a Modal

### PR DESCRIPTION
**Pull request by JasonMHasperhoven**
**Description**

The current settings page is a bit awkward, so thought to make it similar to Slack's settings modal with tabs for each section. Did a bit of cleanup on the UI as well, including feedback if the api keys are configured.

image

image

image

### CODE REVIEW

File: src/components/AppSettingsModal/AppSettingsModal.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      </div>
+      <div className='appsettings-modal__content'>
+        {activeTab === Tabs.General && <GeneralTab />}
+        {activeTab === Tabs.Keys && <ApiKeysTab />}
+        {activeTab === Tabs.TradingMode && <TradingModeTab onClose={onClose} />}
+      </div>
+    </Modal>
+  )
+}
+
```
Comment (Blaumaus) [inline src/components/AppSettingsModal/AppSettingsModal.js:71]: We should validate the prop types here.

```suggestion

AppSettingsModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
}

```

File: src/components/AppSettingsModal/AppSettingsModal.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable react/prop-types */
```
Comment (Blaumaus) [inline src/components/AppSettingsModal/AppSettingsModal.js:1]: Don't need this comment here anymore; added import for `PropTypes`

```suggestion
import React, { useState } from 'react'
import PropTypes from 'prop-types'
```

File: src/components/AppSettingsModal/AppSettingsModal.TradingMode.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Checkbox, Button, Intent } from '@ufx-ui/core'
+
+import UIActions from '../../redux/actions/ui'
+import { getIsPaperTrading } from '../../redux/selectors/ui'
+
+// eslint-disable-next-line react/prop-types
+const TradingMode = ({ onClose }) => {
```
Comment (Blaumaus) [inline src/components/AppSettingsModal/AppSettingsModal.TradingMode.js:9]: PropTypes here also should be validated to maintain codebase consistency

**Review by Blaumaus — CHANGES_REQUESTED** [review]
This new modal looks amazing! Good job with some minor proposals

**Review by bhoomij — APPROVED** [review]
looking great 👍 

could you add asana task please ? so this is not missed from QA cycle

**Comment by robertkowalski** [comment]
created https://app.asana.com/0/1125859137800433/1200571179851769

**Pull request merged**